### ISSUE TITLE
Provide API key use logs to users

### ISSUE THREAD

**Issue by mcgibbon** [esi/esi-issues#1115]
# Feature Request

To protect users from misuse, the ESI [license agreement](https://developers.eveonline.com/resource/license-agreement) requires that applications do not use their ESI access for the following (Section 2.3):

> (c) as a means of tracking Player information or Player activity without the express knowledge and consent of such Player, (d) as a means to misappropriate a Player's in-game items or other information, or to otherwise cheat, scam, or defraud Players who subscribe to the Application

Currently, there is no way for a user of an application to hold that application accountable to follow this part of the license agreement. Previously, [logs of API access](https://community.eveonline.com/support/api-key/log/) were provided to users for the XML api. I would propose that a similar log be provided for ESI via the support portal they can use to view their granted permissions (and not through ESI itself as previously suggested in #223).

This would enable users to open support tickets when they detect their api keys are being misused, instead of depending on CCP as the only entity which can detect misuse.

## Use case

Any user of an ESI application which asks for api access from that user's characters. Directly, interested users can use this feature to check that their api key is not being misused. Indirectly, less interested users are protected from misuse by the fact that other users can check the application for compliance.

## Authentication

No new scopes, just accessible to a user the same way they do any other api key management.

## Example return

A log page similar to [this one](https://community.eveonline.com/support/api-key/log/), apart from possibly not including the return value (which could get pretty verbose), and including application name instead of IP address. Columns include Date and time, Application name, and "Page & Parameters".

# Checklist

Check all boxes that apply to this issue:

- [x] Feature request description is provided
- [x] Use case exists
- [ ] Feature requires a new route
- [ ] Feature adds data to existing route
- [ ] Feature requires new auth scope
- [ ] Feature can reuse existing scope
- [ ] Feature does not require auth
- [x] Meta feature, applies to all routes

**Comment by ddouglas** [comment]
I have to disagree with this ticket. Let me explain

Prior to ESI, it was easy to abuse the API Authentication Layer because you could give your API Key to a single person or application and those keys could be passed from person to person or from application to application with no control over who was using them to get your data. At that time, it made sense to have a log of who, or rather what device (because if I remember correctly only an IP Address was revealed), was accessing your data. Now the responsibility for that has been shifted to the user by way of the SSO. The user has to authorize the application to use their token and no one else can use that grant. It is the responsibility of the user to ensure that their token is not being abused by ensuring that they have only authorized the necessary scopes and the necessary applications to access their data.

If a user is concerned that an application is abuse their access to their data, then they can easily revoke that applications access and in a little as twenty minutes, their data will be safe. 

To recap, the log you are requesting is already available in the form of the Third Party Application management page.

**Comment by stcktrce** [comment]
> Prior to ESI, it was easy to abuse the API Authentication Layer because you could give your API Key to a single person or application and those keys could be passed from person to person or from application to application with no control over who was using them to get your data.

👎 - I disagree with this. ESI does not change the fact that hostile applications, or badly written applications can leak or share your ESI credentials.

> It is the responsibility of the user to ensure that their token is not being abused by ensuring that they have only authorized the necessary scopes and the necessary applications to access their data.

Even tough the EULA prevents end-users from building shady applications that claim they need your location information so that they can track wormholes, the user will never know that this shady application might actually be polling their location 24x7 and sharing it with others for the purpose of tracking specific players.

Having access to logs that show the usage, frequency and host would allow end-users to determine if an application is doing shady things.

> If a user is concerned that an application is abuse their access to their data, then they can easily revoke that applications access and in a little as twenty minutes, their data will be safe.

I agree with this, but without having access to any of the previously mentioned logs. How will an user know their data is being abused?

**Comment by mcgibbon** [comment]
> To recap, the log you are requesting is already available in the form of the Third Party Application management page.

That's just false - I'm requesting a log of how applications are using their access, not what applications have access.

> If a user is concerned that an application is abuse their access to their data, then they can easily revoke that applications access and in a little as twenty minutes, their data will be safe.

How can a user be concerned that an application abuses their access to data when they have no way to tell how an application is using access to their data? Another user expressed that frustration [here](https://www.reddit.com/r/Eve/comments/7qkz28/eve_applications_postesi_would_you_trust_a_site/dsq0kte?utm_source=share&utm_medium=web2x). The full burden of enforcing the ESI license falls on CCP.

And players don't believe CCP actually does or will spend their time enforcing these rules (on the same page, the very next three comments and many more, [here](https://www.reddit.com/r/Eve/comments/7qkz28/eve_applications_postesi_would_you_trust_a_site/dsq3d55?utm_source=share&utm_medium=web2x), [here](https://www.reddit.com/r/Eve/comments/7qkz28/eve_applications_postesi_would_you_trust_a_site/dsq3d55?utm_source=share&utm_medium=web2x), [here](https://www.reddit.com/r/Eve/comments/7qkz28/eve_applications_postesi_would_you_trust_a_site/dsq3d55?utm_source=share&utm_medium=web2x)). The main nullsec alliances straight up tell their players not to use any applications created by anyone not in their alliance, because there's no way to assess whether that application is complying with the license agreement.

**Comment by mcgibbon** [comment]
You do have a point that there is some information that is still provided from back when the old log page was given, but very important information is no longer given, and that's what this ticket is about.
